src/app/[locale]/onboarding/loading/_hooks/use-logic.ts

```diff
@@ -0,0 +1,180 @@
+"use client";
+
+import { useQuery } from "@tanstack/react-query";
+import i18next from "i18next";
+import { useRouter, useSearchParams } from "next/navigation";
+import { useEffect, useState } from "react";
+import { DEFAULT_PACKAGE_API_URL } from "~/constants";
+import usePackageAPI from "~/hooks/use-package-api";
+import useSQLInit from "~/hooks/use-sql-init";
+import { useAppStore } from "~/stores";
+
+type API = ReturnType<typeof usePackageAPI>;
+
+function useURLParams() {
+  const searchParams = useSearchParams();
+
+  const packageLink = searchParams?.get("packageLink") || undefined;
+  const backendURL = searchParams?.get("backendURL") || undefined;
+  const UPNKey = packageLink
+    ? new URL(packageLink).searchParams.get("upn") || undefined
+    : undefined;
+
+  return { packageLink, backendURL, UPNKey };
+}
+
+function useProcess({
+  api,
+  packageLink,
+  enabled,
+}: {
+  api: API;
+  packageLink: string;
+  enabled: boolean;
+}) {
+  const { data, status } = useQuery({
+    queryKey: ["package-api", "process", packageLink],
+    queryFn: () => api.process({ packageLink }),
+    enabled,
+    staleTime: Infinity,
+  });
+
+  return { data, status };
+}
+
+function useStatus({
+  api,
+  packageID,
+  UPNKey,
+  enabled,
+}: {
+  api: API;
+  packageID: string;
+  UPNKey: string;
+  enabled: boolean;
+}) {
+  const [erroredStatus, setErroredStatus] = useState(false);
+
+  const { data, status } = useQuery({
+    queryKey: ["package-api", "status", packageID],
+    queryFn: () =>
+      api.status({
+        packageID,
+        UPNKey,
+      }),
+    enabled: enabled && !erroredStatus,
+    refetchInterval: 1000, // 1s
+  });
+
+  useEffect(() => {
+    setErroredStatus(data?.isErrored || false);
```

Comment:
```suggestion
    setErroredStatus(!!data?.isErrored);
```

Since other parts use this, I guess it makes sense to use it here as well for consistency